airflow/contrib/operators/dataproc_operator.py

```diff
@@ -205,10 +206,18 @@ def __init__(self,
         self.idle_delete_ttl = idle_delete_ttl
         self.auto_delete_time = auto_delete_time
         self.auto_delete_ttl = auto_delete_ttl
+        self.single_node = num_workers == 0
 
         assert not (self.custom_image and self.image_version), \
             "custom_image and image_version can't be both set"
 
+        assert (
```

Comment:
This assertion does not make any sense now, right? Since `single_node = num_workers == 0`, the second one will statement always be True.

Reply:
I can imagine someone trying to setup cluster which consists only preemptible workers (`num_preemptible_workers`) and no regular workers (`num_workers`). 

Generally, we could assert on `num_workers` and `num_preemptible_workers` and allow only cases: 
- both values are zeros
- `num_workers` >= 2

WDYT?

Reply:
Oh, that is true! ;)